app/services/merge_collection.rb

```diff
@@ -0,0 +1,221 @@
+      end
+
+      if !w.representative_version.published?
+        errors << "Work-#{canonical_work.id} is not published"
+      end
+
+      validate_access_controls(w)
+      validate_work_version_metadata(w)
+      validate_creators(w)
+    end
+  end
+
+  private
+
+    def collection
+      @collection ||= Collection.find_by(uuid: uuid)
+    end
+
+    def canonical_work
+      @canonical_work ||= collection.works.first
+    end
+
+    def canonical_work_version
+      @canonical_work_version ||= canonical_work.versions.first
+    end
+
+    def paper_trail_user
+      @paper_trail_user ||= canonical_work_version.versions.first&.whodunnit
+    end
+
+    def validate_access_controls(work)
+      %i[discover read edit].each do |level|
+        %i[users groups].each do |agent|
+          if canonical_work.send("#{level}_#{agent}").map(&:id) != work.send("#{level}_#{agent}").map(&:id)
+            w_access = work.send("#{level}_#{agent}").inspect
+            canonical_access = canonical_work.send("#{level}_#{agent}").inspect
+            diff = "#{canonical_access}\n#{w_access}"
+
+            errors << "Work-#{canonical_work.id} has different #{level} #{agent} than Work-#{work.id}\n#{diff}"
+          end
+        end
+      end
+    end
+
+    def validate_creators(work)
+      if canonical_work_version.creators.map(&:actor_id) != work.representative_version.creators.map(&:actor_id)
+        w_creators = work.representative_version.creators.inspect
+        canonical_creators = canonical_work_version.creators.inspect
+        diff = "#{canonical_creators}\n#{w_creators}"
+
+        errors << "Work-#{canonical_work.id} has different creators than Work-#{work.id}\n#{diff}"
+      end
+    end
+
+    def validate_work_version_metadata(work)
+      work_version_metadata_diff = MetadataDiff.call(
+        canonical_work_version,
+        work.representative_version
+      )
+
+      attributes_we_dont_care_about = %i[title]
+      work_version_metadata_diff = work_version_metadata_diff.reject do |k, _v|
+        attributes_we_dont_care_about.include?(k.to_sym)
+      end
+
+      if work_version_metadata_diff.any?
+        printable_diff = work_version_metadata_diff.inspect
+        errors << "Work-#{canonical_work.id} has different WorkVersion metadata than Work-#{work.id}\n#{printable_diff}"
+      end
+    end
+
+    def merge_collection
+      deposited_at = collection
+        .works
+        .map(&:deposited_at)
+        .compact
+        .min
+
+      all_file_resources = collection
+        .works
+        .map(&:representative_version)
+        .map(&:file_resources)
+        .flatten
+
+      work = Work.build_with_empty_version(
+        work_type: canonical_work.work_type,
+        embargoed_until: canonical_work.embargoed_until,
+        doi: canonical_work.doi,
+        depositor: canonical_work.depositor,
+        proxy_depositor: canonical_work.proxy_depositor,
+        deposited_at: deposited_at
```

Comment:
@nicktiberi  is merging mostly gonna be for older collections? if for newer as well, maybe need to add the new column `notify_editors` to get the value from canonical work as well, that PR just merged. 🤔

Reply:
Good call - this will be mostly for older works, but we should be thorough and validate/merge all fields that are part of the canonical work.